src/lib.rs

```diff
@@ -277,3 +259,57 @@ impl State {
         }
     }
 }
+
+fn join_users_to_room(
```

Comment:
I believe this function must return a `Future` instead of spawning a tokio task, then the `buffer_unordered` will handle the completion of the futures using a `FuturesUnordered` internally.

Reply:
Oh right, that makes sense, will remove the tokio spawn